### ISSUE TITLE
options.dirName and serving static content

### ISSUE THREAD

**Issue by dschulten** [smollweide/node-mock-server#42]
I am looking for a possibility to serve static content (html, css, js, images) not via the rest api definitions, but just from an arbitrary folder. Right now I am serving static content through a separate server on localhost on a different port, but that makes things more complicated since I need CORS.

I was hoping serving static content could be done with options.dirName. It is documented like this:

options.dirName
Type: String
A string value that defines the root directory (__dirname).

I tried 
options.dirName: __dirname
options.dirName:  __dirname + '/public
options.dirName: __dirname + '/../public

and put a file foo.html to __dirname (the folder where index.js resides), __dirname/public and a /public folder right beside __dirname, but I was never able to get the http://localhost:3003/foo.html. Rather I got the usual error response for a missing rest resource definition:

Error: ENOENT: no such file or directory, open 'C:\foo.html\#\OPTIONS\mock\response.txt'

Two questions arise:
Is it possible to serve arbitrary static content through node-mock-server?
If options.dirName does not define the folder where static content is served from, what is its purpose?

**Comment by smollweide** [comment]
`options.dirName: __dirname` is correct.
Static content is not yet supported but also a good point.

**Comment by pcaithness** [comment]
I have the same requirement.
All I did was start up a simple python http-server on another port. My app config allows me to specify a static url endpoint thats different to my api endpoint.

**Comment by dschulten** [comment]
I just saw your commit and realized that you are implementing here what I intended in #44 (ReST Responses that are not JSON)  That is also something I need :), but to clarify my intention for #42:

What I meant here is that I need one or more folders which do not become part of the mock api, but are served as the actual SPA (and I think @pcaithness needs that as well). To illustrate, _/public_ and _dist_ would be those spa folders in the project below:
```
/src      <-- js and scss
/public <-- node-mock-server serves this content statically, possibly at localhost:3003/
  index.html
/dist     <-- node-mock-server serves this content statically, also at localhost:3003/
  bundle.js
  bundle.css
  logo.gif
/mock <-- node-mock-server serves this as mock ReST api
  /func
  /rest
```
I just realized that this conflicts with the fact that the node-mock-server ui is served at _localhost:3003/_ right now. One possible solution could be to serve the node-mock-server ui at _localhost:3003/node-mock-server_. Or the public and dist folder could be served at _localhost:3003/my-app-name_ instead (although I would prefer to be able to serve my app at / and rather have the node-mock-server ui at a different path)

**Comment by dschulten** [comment]
I have tried out the current develop branch and it still seems work in progress, maybe you _do_ want to take a look at my [branch # 44](https://github.com/dschulten/node-mock-server/tree/%2344). In my soluation you choose `success` and then you can request the resource with Accept: application/json, application/xml, text/html and node-mock-server handles content negotiation.

Or maybe I misunderstood what you are trying to do in the develop branch right now?

**Comment by smollweide** [comment]
Hi @dschulten,
I added the expressMiddleware option with commit 4232c1a.
Solution for your example:
```
expressMiddleware: [
	['/public', express.static(__dirname + '/public')],
	['/dist', express.static(__dirname + '/dist')]
],
```

**Comment by dschulten** [comment]
Looks pretty good. Since I want /public and /dist served at / of my project, I now have:
```
  expressMiddleware: [
    ['/', express.static(__dirname + '/../public')],
    ['/', express.static(__dirname + '/../dist')],
  ],
```
http://localhost:3003 now serves my app, that's great.

However, that makes the mockserver ui unreachable. Can I configure the mockserver ui to run at localhost:3003/ui somehow?

**Comment by dschulten** [comment]
There seems to be yet another issue. tl;dr expressMiddleware requires npm3.

A mockServer script with expressMiddleware needs express as project dependency so that require('express') works. If I add express to the devDependencies of my project to fulfill the requirement: 

    npm i express -D

the ui is still shown. However, when I run npm install (npm2 with node 4.5.0) on the whole project, the `express` dependency gets removed from the node_modules of node-mock-server, and now I get the following error from the UI when requesting localhost:3003:

 ```
Error: Cannot find module 'ejs'
    at Function.Module._resolveFilename (module.js:325:15)
    at Function.Module._load (module.js:276:25)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at new View (C:\AZ_DATEN\ws\ztb-spa\node_modules\express\lib\view.js:80:30)
    at EventEmitter.render (C:\AZ_DATEN\ws\ztb-spa\node_modules\express\lib\application.js:570:12)
    at ServerResponse.render (C:\AZ_DATEN\ws\ztb-spa\node_modules\express\lib\response.js:971:7)
    at UiController.extend._renderUi (C:\AZ_DATEN\ws\ztb-spa\node_modules\node-mock-server\lib\controller\UiController.js:47:7)
    at Layer.handle [as handle_request] (C:\AZ_DATEN\ws\ztb-spa\node_modules\express\lib\router\layer.js:95:5)
    at next (C:\AZ_DATEN\ws\ztb-spa\node_modules\express\lib\router\route.js:137:13)
```
A workaround is to use npm3, which dedupes the express dependency differently and works for me. However, npm3 does not come with older node versions like 4.5.0, so people have to upgrade their npm inside their node installation in order to use expressMiddleware.

Can you solve this without forcing people to upgrade to npm3?

**Comment by smollweide** [comment]
That 59d08acc should fix the npm2 issue.

**Comment by smollweide** [comment]
The commit 7af8e7c makes the UI path configurable

**Comment by smollweide** [comment]
Implemented with release [0.16.0](https://github.com/smollweide/node-mock-server/releases/tag/0.16.0)

**Comment by dschulten** [comment]
thank you very much!
